src/types.ts

```diff
@@ -89,26 +89,40 @@ export interface GetSearchRequest {
   matches?: boolean
 }
 
-export type Hit<T> = T & { _formatted?: T }
-
 export type CategoriesDistribution = {
   [category: string]: number
 }
 
 export type Facet = string
+export type FacetsDistribution = Record<Facet, CategoriesDistribution>
+export type _matchesInfo<T> = Partial<
+  Record<keyof T, Array<{ start: number; length: number }>>
+>
+
+export type Hit<T> = T & {
+  _formatted?: Partial<T> // SHOULD BE IMPROVED !
```

Comment:
Could you please open an issue instead of letting comment in the code? We forget comments in the code 🙂

Reply:
Ah yes I did the issue I forgot to remove the comment: https://github.com/meilisearch/meilisearch-js/issues

Reply:
Awesome!